presto-main/src/main/java/com/facebook/presto/sql/planner/StatisticsAggregationPlanner.java

```diff
@@ -0,0 +1,251 @@
+import com.facebook.presto.sql.planner.plan.StatisticAggregations;
+import com.facebook.presto.sql.planner.plan.StatisticAggregationsDescriptor;
+import com.facebook.presto.sql.tree.Cast;
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.sql.tree.FunctionCall;
+import com.facebook.presto.sql.tree.QualifiedName;
+import com.facebook.presto.sql.tree.SymbolReference;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.AbstractMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.spi.StandardErrorCode.NOT_SUPPORTED;
+import static com.facebook.presto.spi.statistics.TableStatisticType.ROW_COUNT;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.VarbinaryType.VARBINARY;
+import static com.facebook.presto.spi.type.Varchars.isVarcharType;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Verify.verify;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class StatisticsAggregationPlanner
+{
+    private final SymbolAllocator symbolAllocator;
+    private final Metadata metadata;
+
+    public StatisticsAggregationPlanner(SymbolAllocator symbolAllocator, Metadata metadata)
+    {
+        this.symbolAllocator = requireNonNull(symbolAllocator, "symbolAllocator is null");
+        this.metadata = requireNonNull(metadata, "metadata is null");
+    }
+
+    public CreateStatisticAggregationsResult createStatisticsAggregation(TableStatisticsMetadata statisticsMetadata, Map<String, Symbol> columnToSymbolMap)
+    {
+        StatisticAggregationsDescriptor.Builder<Symbol> descriptor = StatisticAggregationsDescriptor.builder();
+
+        List<String> groupingColumns = statisticsMetadata.getGroupingColumns();
+        List<Symbol> groupingSymbols = groupingColumns.stream()
+                .map(columnToSymbolMap::get)
+                .collect(toImmutableList());
+
+        for (int i = 0; i < groupingSymbols.size(); i++) {
+            descriptor.addGrouping(groupingSymbols.get(i), groupingColumns.get(i));
+        }
+
+        ImmutableMap.Builder<Symbol, AggregationNode.Aggregation> aggregations = ImmutableMap.builder();
+
+        for (TableStatisticType type : statisticsMetadata.getTableStatistics()) {
+            if (type != ROW_COUNT) {
+                throw new PrestoException(NOT_SUPPORTED, "Table-wide statistic type not supported: " + type);
+            }
+        }
+
+        FunctionRegistry functionRegistry = metadata.getFunctionRegistry();
+        if (!statisticsMetadata.getTableStatistics().isEmpty()) {
```

Comment:
You should merge this `if` into the `for` loop above:
```
for (TableStatisticType type : statisticsMetadata.getTableStatistics()) {
    if (type != ROW_COUNT) {
       ...
    }

    // plan the row count agg
}
```

Reply:
Merged